### PR TITLE
changes of closing the stream on ioexception

### PR DESCRIPTION
**Pull request by CMahesh**
The followUpRequest IOException is catch and the streamallactor is closed and the exception is re throws the IOException

### CODE REVIEW

File: okhttp/src/main/java/okhttp3/internal/http/RetryAndFollowUpInterceptor.java [hunk 1]
```diff
@@ -155,7 +155,13 @@ public StreamAllocation streamAllocation() {
             .build();
       }
 
-      Request followUp = followUpRequest(response, streamAllocation.route());
+      Request followUp = null;
+      try {
+        followUp = followUpRequest(response, streamAllocation.route());
+      } catch (IOException e) {
+          streamAllocation.release();
```
Comment (yschimke) [inline okhttp/src/main/java/okhttp3/internal/http/RetryAndFollowUpInterceptor.java]: Formatting looks off

**Comment by CMahesh** [comment]
@yschimke  ok sure. let me add a test

**Comment by yschimke** [comment]
Do you need any help with the test?

**Comment by PromanSEW** [comment]
This PR can be closed as of #3971

**Pull request closed without merging**